packages/api/src/util/copyCropPlan.js

```diff
@@ -0,0 +1,639 @@
+              ? {
+                  ..._omit(plan.container_method, getPropertiesToDelete(ContainerMethodModel)),
+                  planting_management_plan_id:
+                    newPlantingManagementPlanUUIDs[plan.planting_management_plan_id],
+                }
+              : null,
+            broadcast_method: plan.broadcast_method
+              ? {
+                  ..._omit(plan.broadcast_method, getPropertiesToDelete(BroadcastMethodModel)),
+                  planting_management_plan_id:
+                    newPlantingManagementPlanUUIDs[plan.planting_management_plan_id],
+                }
+              : null,
+            row_method: plan.row_method
+              ? {
+                  ..._omit(plan.row_method, getPropertiesToDelete(RowMethodModel)),
+                  planting_management_plan_id:
+                    newPlantingManagementPlanUUIDs[plan.planting_management_plan_id],
+                }
+              : null,
+          };
+        },
+      ),
+    },
+    repetition_number: index + 1,
+  };
+};
+
+/**
+ * Generates a management plan group graph based on the provided parameters.
+ *
+ * The function constructs a management plan group graph object which can be upserted
+ * on the management plan group model.
+ *
+ * @param {string} createdByUser - The username of the user who initiated the management plan group repetition.
+ * @param {Object} repeatDetails - The repetition configuration for the management plan group template.
+ * @param {Date[]} sortedStartDates - An array of sorted start dates for the management plan templates.
+ * @param {Object} managementPlanGraph - The management plan graph used as the template object.
+ * @param {Object} theOnlyActiveUserFarm - Null if more than one 'Active' userFarm.
+ * @param {Date|string} firstTaskDate - The date of the first task in the management plan templates.
+ *                                      It can be a Date object or a date string in "YYYY-MM-DD" format.
+ * @returns {Object} - The management plan group graph object to be upserted.
+ */
+export const getManagementPlanGroupTemplateGraph = (
+  createdByUser,
+  repeatDetails,
+  sortedStartDates,
+  managementPlanGraph,
+  theOnlyActiveUserFarm,
+  firstTaskDate,
+) => {
+  return {
+    repetition_count: sortedStartDates.length,
+    repetition_config: repeatDetails,
+    management_plans: sortedStartDates.map((date, index) => {
+      const newPlantingManagementPlanUUIDs = getUUIDMap(
+        managementPlanGraph.crop_management_plan.planting_management_plans,
+        'planting_management_plan_id',
+      );
+      return getManagementPlanTemplateGraph(
+        date,
+        index,
+        repeatDetails,
+        createdByUser,
+        managementPlanGraph,
+        theOnlyActiveUserFarm,
+        firstTaskDate,
+        newPlantingManagementPlanUUIDs,
+      );
+    }),
+  };
+};
+
+/**
+ * Used for jest tests only generates a management plan group graph based on the provided parameters.
```

Comment:
@iperdomo Where are functions for tests usually placed? I am wondering if we should have a util folder under `tests`.

Reply:
Good point, I didn't look to hard for this... But also I do kind of like the idea of all specific feature utils in one place.

Reply:
Agree with @SayakaOno test utility functions should be under `tests/utils/*`

Reply:
@iperdomo @SayakaOno Looking at this further are you sure this is desirable? Utilities are utilities no matter where they are?

Doing this results in the following:

managementPlanTest imports regular copyUtils containing these:
  
![Screenshot 2023-07-27 at 8 34 38 AM](https://github.com/LiteFarmOrg/LiteFarm/assets/30075973/0875f79d-039c-4ed8-8e98-60bc0b41d792)

managementPlanTest now also imports the test utils which imports many of the same imports:
![Screenshot 2023-07-27 at 8 33 02 AM](https://github.com/LiteFarmOrg/LiteFarm/assets/30075973/25a6ea4e-46de-4a45-80f7-db7e3454f120)

test utils also actually needs the regular copy utils to function too (first picture). 

...So we are possibly duplicating these same imports 3 times